Wootric/WootricSDK-Android
# PBS: add support for Property Based Sampling

Add support for Property Based Sampling

## Changes:

- Update eligibility call to include properties
- Update eligibility's response log to show more information
- Add tests

## How to test

This can be tested in production since PBS is already live.

1. Enable PBS for the account you'll use to test this
2. Add a new property sampling rule via the UI
3. Setup SDK

```java

        Wootric wootric = Wootric.init(this, CLIENT_ID, ACCOUNT_TOKEN);
        wootric.setEndUserEmail("pbs@test.com");
        wootric.setEndUserCreatedAt(1544659200);
        HashMap<String, String> properties = new HashMap<>();
        properties.put("pricing_plan", "pro plan");
        wootric.setProperties(properties);
        wootric.survey();
```

4. In Android Studio you can see the logs whit the call to eligibility with the properties as params
5. After responding to the survey, you should see the new response with in the dashboard. You'll also see a new option inside Segments & Filters: 'Wootric Sampling Rule' click on it and the name of your rule will be in there.
 
## Trello
https://trello.com/c/bCFWiYRN/3964-add-pbs-to-mobile-sdks
